app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -211,64 +222,61 @@ private function parseReferralCode($referralCode)
     }
 
     /**
-     * For a given record and referral code values, first check if we have a northstar ID, then grab the user using that.
-     * Otherwise, see if we can find the user with the given email (if it exists), if not check if we can find them with the given phone number (if it exists).
-     * If all else fails, create the user .
+     * Fetch user ID by given email or mobile if exists. If not found, create new user.
      *
-     * @TODO - If we have a northstar id in the referral code, then we probably don't need to make a call to northstar for the full user object.
-     *
-     * @return array
+     * @return string
      */
-    private function getOrCreateUser($record, $values)
+    private function getOrCreateUserId($record)
     {
         $user = null;
+        $recordEmail = $record['Email address'];
+        $recordMobile = $record['Phone'];
+
+        if (isset($recordEmail) && !empty($recordEmail)) {
+            info('fetching user by email '.$recordEmail);
+            $user = gateway('northstar')->asClient()->getUser('email', $recordEmail);
+        } else {
+            info('fetching user by mobile '.$recordMobile);
+            $user = gateway('northstar')->asClient()->getUser('mobile', $recordMobile);
+        }
 
-        $userFieldsToLookFor = [
-            'id' => isset($values['northstar_id']) && !empty($values['northstar_id']) ? $values['northstar_id'] : null,
-            'email' => isset($record['Email address']) && !empty($record['Email address']) ? $record['Email address'] : null,
-            'mobile' => isset($record['Phone']) && !empty($record['Phone']) ? $record['Phone'] : null,
-        ];
+        if (!is_null($user)) {
+            return $user->id;
+        }
 
-        foreach ($userFieldsToLookFor as $field => $value)
-        {
-            if ($value) {
-                info('getting user with the '.$field.' field', [$field => $value]);
-                $user = gateway('northstar')->asClient()->getUser($field, $value);
-            }
+        $userData = [
+            'email' => $recordEmail,
+            'mobile' => $recordMobile,
+            'first_name' => $record['First name'],
+            'last_name' => $record['Last name'],
+            'addr_street1' => $record['Home address'],
+            'addr_street2' => $record['Home unit'],
+            'addr_city' => $record['Home city'],
+            'addr_state' => $record['Home state'],
+            'addr_zip' => $record['Home zip code'],
+            'source' => env('NORTHSTAR_CLIENT_ID'),
```

Comment:
Let's read this from `config('services.northstar.client_credentials.client_id')` instead so we don't prevent ourselves from using [config caching](https://laravel.com/docs/5.7/configuration#configuration-caching) here (if we don't already).